### PR TITLE
Support for Tracker Exceptions

### PR DESCRIPTION
**Pull request by bwaresiak**
Task/Issue URL: https://app.asana.com/0/856498667320406/1200610585462642
Tech Design URL:
CC:

**Description**:
Support for Tracker Exceptions used in case blocking given tracker causes critical site breakage.

**Steps to test this PR**:

- Migration from old rules identifier to new one for existing users.
- Test on privacy test pages.
- Check unit tests for possible mistakes/missing functionality.
- General smoke tests (toggling protection, fire button, surrogates, etc).

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPad

**OS Testing**:

* [ ] iOS 13
* [ ] iOS 14
* [ ] iOS 15

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**

### CODE REVIEW

File: Core/contentblockerrules.js [hunk 1]
```diff
@@ -43,6 +43,102 @@
...
+
+            if (url.startsWith("//")) {
+                url = "http:" + url;
+            }
+
+            try {
+                var parsed = new URL(url);
+                return {
+                    domain: parsed.hostname,
+                    hostname: parsed.hostname
```
Comment (SlayterDev) [inline Core/contentblockerrules.js:57]: Are these meant to be the same property?

**Review by SlayterDev — COMMENTED** [review]
Overall looks good. Left one comment.

Additionally, I noticed the changes in the .js files are largely the same. Is it possible to extract this logic out to a single script and either prepend it when creating the scripts or inject it as a replacement? The idea being to cut down on code duplication.

File: Core/contentblockerrules.js (hunk 1 above)
Comment (bwaresiak) [reply Core/contentblockerrules.js:57]: That's how it's been in https://github.com/duckduckgo/iOS/blob/develop/Core/contentblocker.js - I just coped the code.

**Review by mallexxx — APPROVED** [review]
Code changes look good to me! Can't wait for this to be shared in BSK
Although you may want somebody from the iOS team to validate this running on-device

**Review by Bunn — APPROVED** [review]
Migrations are running correctly, no unecessary calls to compile being made.
Privacy test pages verified 
Tests are running correctly after submodule update
Fire button / toggling protection working as expected

Tested on iPhone SE (iOS 14) and iPhone 6S (iOS 15)

LGTM!

**Pull request merged**